src/pages/worklist/DOIRequestItem.tsx

```diff
@@ -0,0 +1,199 @@
+import React, { FC, useState } from 'react';
+import styled from 'styled-components';
+import { Colors, DeviceWidths, StyleWidths } from '../../themes/mainTheme';
+import { WorklistDOIRequest } from '../../types/Worklist.types';
+import {
+  Button,
+  CircularProgress,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+  Grid,
+  Link,
+  TextField,
+  Typography,
+  useMediaQuery,
+} from '@material-ui/core';
+import DeleteIcon from '@material-ui/icons/Delete';
+import { format } from 'date-fns';
+import { useTranslation } from 'react-i18next';
+import { createDOI, refuseDoiRequest } from '../../api/workListApi';
+import ErrorBanner from '../../components/ErrorBanner';
+
+interface Props {
+  backgroundColor: string;
+}
+
+const StyledListItemWrapper: any = styled.li<Props>`
+  width: 100%;
+  max-width: ${StyleWidths.width5};
+  background-color: ${(props: any) => props.backgroundColor || Colors.UnitTurquoise_20percent};
+  padding: 1rem;
+  display: flex;
+  justify-content: center;
+  margin-top: 1rem;
+`;
+
+interface DOIRequestItemProps {
+  workListRequestDOI: WorklistDOIRequest;
+  setWorkListDoi: React.Dispatch<React.SetStateAction<WorklistDOIRequest[]>>;
+}
+
+const DOIRequestItem: FC<DOIRequestItemProps> = ({ workListRequestDOI, setWorkListDoi }) => {
+  const [isBusy, setIsBusy] = useState(false);
+  const [showLongText, setShowLongText] = useState(false);
+  const [showConfirmDeleteDialog, setShowConfirmDeleteDialog] = useState(false);
+  const [updateError, setUpdateError] = useState<Error>();
+  const [deleteComment, setDeleteComment] = useState('');
+  const fullScreenDialog = useMediaQuery(`(max-width:${DeviceWidths.sm}px)`);
+  const { t } = useTranslation();
+
+  const deleteRequest = async (ResourceIdentifier: string, comment: string) => {
+    try {
+      setIsBusy(true);
+      setUpdateError(undefined);
+      await refuseDoiRequest(ResourceIdentifier, comment);
+      setWorkListDoi((prevState) => prevState.filter((work) => work.resourceIdentifier !== ResourceIdentifier));
+    } catch (error) {
+      setUpdateError(error);
+    } finally {
+      setIsBusy(false);
+    }
+  };
+
+  const contactApiForDoi = async (ResourceIdentifier: string) => {
+    try {
+      setIsBusy(true);
+      setUpdateError(undefined);
+      await createDOI(ResourceIdentifier);
```

Comment:
hva med en feedback-alert for create-doi ?

Reply:
Er det ikke greit uten når listItemet forsvinner?

Reply:
kanskje bedre om vi har en confirmation-dialog først, ja